File: guanabara-java/src/com/danillo/algoritmos/lista05/Exercicio59.java
package com.danillo.algoritmos.lista05;

import java.util.Scanner;

public class Exercicio59 {
    public static void main(String[] args) {
        Scanner entrada = new Scanner(System.in);

        int maiorIdade = 0, homensCadastrados = 0, mulherMaisJovem = Integer.MAX_VALUE, somaIdadeHomens = 0;
        char continuar = 'S';

        while(continuar == 'S'){
            System.out.println("Insira a idade: ");
            int idade = entrada.nextInt();
            System.out.println("Insira o gênero: ");
            char genero = entrada.next().charAt(0);

            if(idade > maiorIdade){
                maiorIdade = idade;
            }

            if (genero == 'M'){
                somaIdadeHomens = somaIdadeHomens + idade;
                homensCadastrados++;
            }

            if (genero == 'F' && idade < mulherMaisJovem){
                mulherMaisJovem = idade;
            }

            System.out.println("Deseja continuar? (S ou N)");
            continuar = entrada.next().charAt(0);
        }

        float mediaIdadeHomens = (float)somaIdadeHomens/homensCadastrados;

        System.out.println("Maior idade: " +maiorIdade+
                           "\nQuantidade de homens cadastrados: " +homensCadastrados+
                           "\nIdade da mulher mais jovem: " +mulherMaisJovem+
                           "\nMédia de idade entre os homens:" +mediaIdadeHomens);
    }
}

File: guanabara-python/src/lista05/exercicio59.py
continuar = 1
maior_idade = 0
homens_cadastrados = 0
soma_idade_homens = 0
idade_mulher_mais_jovem = float('inf')

while continuar == 1:
    idade = int(input("Insira a idade: "))
    genero = int(input("Insira o gênero: 1- Masculino 2 - Feminino "))

    if idade > maior_idade:
        maior_idade = idade

    if genero == 1:
        homens_cadastrados = homens_cadastrados + 1
        soma_idade_homens = soma_idade_homens + idade

    if genero == 2 and idade < idade_mulher_mais_jovem:
        idade_mulher_mais_jovem = idade

    continuar = int(input("Deseja continuar: 1 - Sim | 2 - Não "))

media_idade_homens = soma_idade_homens / homens_cadastrados

print(f"Maior idade: {maior_idade}.\nHomens cadastrados: {homens_cadastrados}.\n"
      f"Idade da mulher mais jovem: {idade_mulher_mais_jovem}.\nMédia de idade dos homens: {media_idade_homens}.")
